utils/imodels-client-test-utils/src/test-context-providers/imodel/TestIModelRetriever.ts

```diff
@@ -104,4 +107,19 @@ export class TestIModelRetriever {
 
     return locks[0];
   }
+
+  private async ensureChangesetGroups(iModelId: string): Promise<ChangesetGroupMetadata[]> {
+    const getChangesetGroupListParams: GetChangesetGroupListParams = {
+      authorization: this._testAuthorizationProvider.getAdmin1Authorization(),
+      iModelId
+    };
+    let changesetGroups: ChangesetGroupMetadata[] = await toArray(this._iModelsClient.changesetGroups.getList(getChangesetGroupListParams));
+
+    if (changesetGroups.length === 0)
+      changesetGroups = await this._testIModelCreator.createChangesetGroups(iModelId);
```

Comment:
we do not create entities on the reusable iModel if we found one but it does not match our expectations. It means that we should recreate it.
![Uploading image.png…]()

Reply:
Hm, what do you mean by recreating? Should we catch the setup error, which is an expected case in upgrade scenario? Otherwise, we need to recreate the iModel. As I'm aware, the easiest way to recreate the iModel with the current implementation:

1. Change the `.env` file for a single run
2. Catch the error and act accordingly

Reply:
> Change the .env file for a single run
this. what we usually do is if we make changes to the reusable iModel creation logic, we set the `recreate` flag to true, run the tests, and that run creates the new iModel. Test run with changes to the reusable iModel and without the `recreate` flag set to true should fail because it likely means an unintentional change to the reusable iModel, this is why why have a bunnch of assertions when we query the reusable iModel
![image](https://github.com/iTwin/imodels-clients/assets/70565417/499bbd55-6ffe-4723-a70a-2425bb363568)